src/Microsoft.ML.Onnx/TransformerChainOnnxConverter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.UniversalModelFormat.Onnx;
+
+namespace Microsoft.ML.Model.Onnx
+{
+    public class TransformerChainOnnxConverter
+    {
+        public static ModelProto Convert<T>(TransformerChain<T> chain, IDataView inputData, HashSet<string> inputColumnNamesToDrop=null, HashSet<string> outputColumnNamesToDrop=null) where T : class, ITransformer
```

Comment:
>TransformerChain<T> chain [](start = 44, length = 25)

Is there a reason to do this? If you wanted specialized behavior for the transformer chain, would it be sufficient to just check if it is a transformer chain in the other method, and not make this public?

The presence of a specialized overload for transformer chain specifically makes me suspect that you're having different behavior for that. But this makes me wonder, since a `TransformerChain<T>` is also an `ITransformer`, does this mean that if I were to just feed it in as such below, that it would not work, or have a flaw of some kind? In which case, the method below we'd say is buggy. (Since it purports to work over any `ITransformer`. But if it *does* work, then why have a separate method anyway? So I'd put that logic (if indeed any must exist) on the second method, and *not* expose whatever implementation specific type polymorphism to the user. Since they shouldn't have to care about that.

Reply:
I will delete the version for `TransformerChain<T>`.

---
In reply to: [245214766](https://github.com/dotnet/machinelearning/pull/2013#discussion_r245214766) [](ancestors = 245214766)